lib/auth/api.go

```diff
@@ -663,6 +663,67 @@ type WindowsDesktopAccessPoint interface {
 	accessPoint
 }
 
+// ReadDiscoveryAccessPoint is a read only API interface implemented by a certificate authority (CA) to be
+// used by a teleport.ComponentDiscovery.
+//
+// NOTE: This interface must match the resources replicated in cache.ForDiscovery.
+type ReadDiscoveryAccessPoint interface {
```

Comment:
Does discovery service actually need all of these? Can we keep only those that it will actually use?